cargo-quickinstall/src/main.rs

```diff
@@ -339,6 +339,11 @@ fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
         None => get_target_triple()?,
     };
 
+    if options.dry_run {
+        println!("[Dry run] curl --user-agent \"cargo-quickinstall client (alsuren@gmail.com)\" --location --silent --show-error --fail https://github.com/alsuren/cargo-quickinstall/releases/download/{crate_name}-{version}-{target}/{crate_name}-{version}-{target}.tar.gz | tar -xzvvf - -C ~/.cargo/bin", crate_name=crate_name, version=version, target=target);
```

Comment:
It's all ended up on one line, so I will write things out as tickboxes. In general, try to avoid writing long lines like this. You can use backslash to escape the newline and subsequent indent if you need to (I can't find the docs for it, but https://stackoverflow.com/a/29483453)

- [ ] could we run the curl command (or with the --head argument to curl added as well) to check that the package is available before printing this?
- [ ] it is possible to make Command print itself in a way that is safe to paste into bash (using its Debug implementation). Could we refactor `curl_bytes(url)` to give us a `curl_command(url)` helper function. This would allow us to share more code (see https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ec2ba6d47b709a065140c8bee4b773b4 for an example)? The tar command might be more difficult to share code for though (see below).
- [ ] (optional) Is `~/.cargo/bin` guaranteed to be where the user's cargo bin lives (even on windows)? That said, my main target audience for this feature is people who want to paste this into their CI pipelines (see the install instructions for https://github.com/cargo-quick/kache as an example). I'm expecting these users to be able to work out where they need to untar to, and this has the potential to add quite a lot of complexity, so I'm happy to leave this as it is.

I realise that I didn't write up this ticket very well. Sorry about that. Feel free to push back against all of those points and I'll happily merge this as-is.

Reply:
I have implemented the above.  Please take a look.

Reply:
It's a thing of beauty